src/main/webapp/creatives/maps/simid_map_creative.test.js

```diff
@@ -22,44 +23,70 @@ test('testing button text updates from ad params', () => {
         timestamp: 0,
         type: "SIMID:Player:init",
     };
-    testMap.onInit(eventData);
+    return eventData;
+}
 
-    const startData = {
+beforeEach(() => {
+    window.google = {
+        maps: {
+            LatLng: jest.fn(), 
+            Map: jest.fn(),
+            Marker: jest.fn(),
+            Autocomplete: class {}
+        }
+      };
+    testMap = new SimidMapCreative();
+    document.body.innerHTML = `
+    <button id="findNearest"></button>
+    <div id="map"></div>`;
+    startData = {
         messageId: 1,
         sessionId: "test-session-id",
         timestamp: 0,
         type: "SIMID:Player:startCreative",  
     }
+});
+
+test('testing button text updates from ad params', () => {
+    const eventData = createInitData('"buttonLabel": "Place"');
```

Comment:
minor nit but single quotes inside a double quote might be better here

Reply:
good call